### PR TITLE
Add pre-commit hooks

### PR DESCRIPTION
**Pull request by bourque**
# Change Summary

## Overview
This PR adds a bunch of `pre-commit-hooks` to match that of the `imap_processing` repo, plus the addition of the `no-commit-to-branch` hook, which should prevent direct commits to the `main` and `development` branches.

## Updated Files
- `.pre-commit-config.yaml`
   - Updated `pre-commit` config file to include the new hooks

### CODE REVIEW

**Comment by bourque** [comment]
@greglucas I figured that I should add the pre-commit hooks that we use under `imap_processing` here too to be consistent with that repo.  I am happy to make changes if this shouldn't be the case.

**Comment by bourque** [comment]
pre-commit.ci autofix

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -3,6 +3,17 @@ ci:
...
+    rev: v4.4.0
+    hooks:
+      - id: check-added-large-files
+      - id: detect-aws-credentials
+        args: [ --allow-missing-credentials ]
+      - id: detect-private-key
+      - id: mixed-line-ending
+      - id: trailing-whitespace
+      - id: no-commit-to-branch
+        args: [--branch, main, --branch, development]
```
Comment (greglucas) [inline .pre-commit-config.yaml]:
```suggestion
        args: [--branch, main, --branch, dev]
```
Can we just switch over to `dev`? That seemed to be the consensus in the Slack thread and is consistent with `imap_processing`.

**Review by greglucas — APPROVED** [review]
I agree, we should try and keep these in-sync and use the same checks in both of them.

**Comment by bourque** [comment]
pre-commit.ci autofix

**Comment by bourque** [comment]
pre-commit.ci autofix

**Comment by bourque** [comment]
I am having the darnedest time trying to get this trailing whitespace pre-commit hook to pass 😅

**Comment by greglucas** [comment]
> I am having the darnedest time trying to get this trailing whitespace pre-commit hook to pass 😅

locally you should be able to try `ruff --fix .` It looks like pre-commit might be queued up in a backlog... Or there aren't many, so maybe just manually fix them up?

**Comment by bourque** [comment]
> > I am having the darnedest time trying to get this trailing whitespace pre-commit hook to pass 😅
> 
> locally you should be able to try `ruff --fix .` It looks like pre-commit might be queued up in a backlog... Or there aren't many, so maybe just manually fix them up?

Thanks @greglucas.  I was able to fix these by doing `pre-commit run --all-files` locally.

**Pull request merged**